### PR TITLE
11 localized resource urls

### LINKED ISSUE #11: Resource URL localized
The resources URLs are not localized (both for the IT and DE the same URL is used). We would like instead to be able to localize them, in order to avoid duplication of resources in the same dataset.

A reference dataset containing the same PDF resource for each supported language can be found at the URL below:

http://dati.retecivica.bz.it/it/dataset/catalogo-beni-culturali

Note: 
A solution could be the management of a simple localized URL for external resources but, for internal resources, we should to expect to upload documents in multiple languages and list only the resources corresponding to the Ckan's locale set.

This work should be implemented as ckanext-multilang enhancement

### PR DESCRIPTION
**Pull request by cezio**
fix #11 
This requires https://github.com/geosolutions-it/ckanext-dcatapit/pull/144

### CODE REVIEW

File: ckanext/multilang/templates/package/snippets/resource_item_multilang.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{% extends 'package/snippets/resource_item.html' %}
+
+
+{% block resource_item_description %}
+    {{ super() }}
+
+    {% if res.lang %}
+        <p> {% trans %}Available in language{% endtrans %}: <span class="text text-info small">{{ h.get_lang_name(res.lang) }} [{{ res.lang }}]</span></p>
```
[the remark this answers is not in the document]
Comment (tdipisa) [reply ckanext/multilang/templates/package/snippets/resource_item_multilang.html]: @cezio why extract_messages did not work correclty?

[the remark this answers is not in the document]
Comment (cezio) [reply ckanext/multilang/templates/package/snippets/resource_item_multilang.html]: No idea why. Maybe extractor didn't scan all directories. I used setup.cfg similar to ones used in other extensions, so I assumed it will work it this case too.

[the remark this answers is not in the document]
Comment (tdipisa) [reply ckanext/multilang/templates/package/snippets/resource_item_multilang.html]: @cezio now if I'm a guest I cannot see the resources

res_pr

[the remark this answers is not in the document]
Comment (tdipisa) [reply ckanext/multilang/templates/package/snippets/resource_item_multilang.html]: @cezio we should also allow a default empty value for the languange selector in resource form

File: ckanext/multilang/plugin.py [hunk 2]
```diff
@@ -287,3 +291,94 @@ def after_create(self, context, resource):
                 ResourceMultilang.persist(resource, lang)
 
 
+LOCALIZED_RESOURCES = 'ckanext.multilang.localized_resources'
+
+class MultilangResourcesPlugin(plugins.SingletonPlugin, toolkit.DefaultDatasetForm):
+
+    ML_ENABLED = 'multilang' in (config.get('ckan.plugins') or '').split(' ')
+    LOCALIZED_RESOURCES_ENABLED = toolkit.asbool(config.get(LOCALIZED_RESOURCES))
+    LOCALIZED_RESOURCES_IN_DCATAPIT = 'dcatapit_pkg' in (config.get('ckan.plugins') or '').split(' ')
```
Comment (etj) [inline ckanext/multilang/plugin.py]: There's a reference to a less general plugin (dcatapit) in a more general one (multilang).
This is a Bad Thing: What if we need multilang in another plugin which conflict? Shall we add exceptions here as well?
Please consider using a configuration line which lists "blacklisted" plugins

File: ckanext/multilang/i18n/de/LC_MESSAGES/ckanext-multilang.po [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+msgstr ""
+
+#: ckanext/multilang/plugin.py:344
+msgid "Set language for which this resource will be visible"
+msgstr ""
+
+#: ckanext/multilang/controllers/package.py:175
+#: ckanext/multilang/controllers/package.py:262
+#, python-format
+msgid "Unauthorized to read package %s"
```
Comment (etj) [inline ckanext/multilang/i18n/de/LC_MESSAGES/ckanext-multilang.po]: shouldnt "{}" be used for params?

File: ckanext/multilang/i18n/ckanext-multilang.pot [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+msgstr ""
+
+#: ckanext/multilang/plugin.py:344
+msgid "Set language for which this resource will be visible"
+msgstr ""
+
+#: ckanext/multilang/controllers/package.py:175
+#: ckanext/multilang/controllers/package.py:262
+#, python-format
+msgid "Unauthorized to read package %s"
```
Comment (etj) [inline ckanext/multilang/i18n/ckanext-multilang.pot]: shouldnt "{}" be used for params?

File: ckanext/multilang/templates/package/edit_multilang.html [hunk 5]
```diff
@@ -0,0 +1 @@
+{% extends "package/edit.html" %}
```
Comment (etj) [inline ckanext/multilang/templates/package/edit_multilang.html]: Is this file really needed? It only extends, but no overrides are implemented

File: ckanext/multilang/i18n/ckanext-multilang.pot (hunk 4 above)
Comment (cezio) [reply ckanext/multilang/i18n/ckanext-multilang.pot]: Those messages are generated from code not covered by this PR: https://github.com/cezio/ckanext-multilang/blame/11_localized_resource_urls/ckanext/multilang/controllers/package.py#L165-L175

File: ckanext/multilang/i18n/de/LC_MESSAGES/ckanext-multilang.po (hunk 3 above)
Comment (cezio) [reply ckanext/multilang/i18n/de/LC_MESSAGES/ckanext-multilang.po]: The same as above, messages are not from my code.

File: ckanext/multilang/plugin.py (hunk 2 above)
Comment (cezio) [reply ckanext/multilang/plugin.py]: True, this is dirty hack on IDatasetForm limitation. The purpose was to have localized resource urls working for multilang standalone and multilang+dcatapit combined deployments. I didn't know any other use cases, hence dcatapit referenced directly.

I'd opt for white-list approach instead, and list extensions supported. If extension is in WL, we assume it will take over IDatasetForm implementation, otherwise ML extension is responsible for this.

File: ckanext/multilang/templates/package/edit_multilang.html (hunk 5 above)
Comment (cezio) [reply ckanext/multilang/templates/package/edit_multilang.html]: Yep, it seems it's unused. Removing..

**Comment by etj** [comment]
Merging as is; opened #14 to clean up unwanted references.
